mlflow/tracking/fluent.py

```diff
@@ -146,6 +147,13 @@ def start_run(run_id=None, experiment_id=None, run_name=None, nested=False):
         if run_name is not None:
             user_specified_tags[MLFLOW_RUN_NAME] = run_name
 
+        if _active_run_stack:
+            user_specified_tags[MLFLOW_ROOT_RUN_ID] = _active_run_stack[0].data.tags[MLFLOW_ROOT_RUN_ID]
+        else:
+            user_specified_tags[MLFLOW_ROOT_RUN_ID] = uuid.uuid4().hex
```

Comment:
```suggestion
            user_specified_tags[MLFLOW_ROOT_RUN_ID] = uuid.uuid4().hex
```

We want to be able to look up all runs that are descendants of a root run based on its ID, so this needs to be equal to the ID of the created run if we're creating a top-level run. Since the run ID is generated by the server at run creation time (as part of handling a CreateRun request), we must instead get the run ID from `active_run_obj` below https://github.com/mlflow/mlflow/pull/2592/files#diff-581b8f8c85a274ab9d0317aec207b364R159 and set it as a tag on the newly-created run